kore/src/Kore/Variables/Fresh.hs

```diff
@@ -38,6 +41,12 @@ class Ord variable => FreshVariable variable where
         -> variable        -- ^ variable to rename
         -> Maybe variable
 
+instance FreshVariable variable => FreshVariable (UnifiedVariable variable)
+  where
+    refreshVariable avoid = traverse (refreshVariable avoid')
+      where
+        avoid' = Set.map asVariable avoid
```

Comment:
I think it's safe because we're avoiding more if we flatten them all to variables. In addition, the fact that their names are anyway distinct should help, too.

Reply:
I've created instances for ElementVariable and SetVariable and changed this to project on the corresponding sets.